### PR TITLE
feature/SOF-3765

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: src/components/ThreejsEditorModal.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+import {ShowIf} from "./ShowIf";
+import settings from "../settings";
+import ModalDialog from "./ModalDialog";
+import {materialsToThreeDSceneData} from "../utils";
+import {LoadingIndicator} from "./LoadingIndicator";
+
+import {THREE_D_BASE_URL, THREE_D_SOURCES} from "../enums";
+
+export class ThreejsEditorModal extends ModalDialog {
```
Comment (timurbazhirov) [inline src/components/ThreejsEditorModal.js]: Let's isolate the logic for loading Three scripts to a separate barebones component and then reuse it in the Modal.

File: src/components/ThreejsEditorModal.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    injectScripts() {
+        THREE_D_SOURCES.forEach(src => {
+            const script = document.createElement("script");
+            script.src = `${THREE_D_BASE_URL}/${src}`;
+            script.async = false;
+            script.defer = false;
+            if (src.includes("SetSceneCommand")) {
+                script.onload = () => {
```
Comment (timurbazhirov) [inline src/components/ThreejsEditorModal.js]: I suggest splitting the function below into domain-focused parts: eg. prepare, load listeners, load materials.

File: src/components/ThreejsEditorModal.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        this.state = {areScriptsLoaded: false};
+        this.injectScripts();
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        window.localStorage.removeItem("threejs-editor");
+    }
+
+    injectScripts() {
+        THREE_D_SOURCES.forEach(src => {
```
Comment (timurbazhirov) [inline src/components/ThreejsEditorModal.js]: The logic used inside this function needs a fair amount of explanation, perhaps in the docstring and/or inline comments

Comment (mohammadidinani) [reply src/components/ThreejsEditorModal.js]: Done.

File: src/components/ThreejsEditorModal.js (hunk 2 above)
Comment (mohammadidinani) [reply src/components/ThreejsEditorModal.js]: Done.

File: src/components/ThreejsEditorModal.js (hunk 1 above)
Comment (mohammadidinani) [reply src/components/ThreejsEditorModal.js]: It is already like that. All the functionality in here is about loading the threejs editor into dom.

**Pull request merged**